diff --git a/APMrover2/Rover.h b/APMrover2/Rover.h
index e2224ea..d62203c 100644
--- a/APMrover2/Rover.h
+++ b/APMrover2/Rover.h
@@ -93,12 +93,19 @@
 #include "Parameters.h"
 #include <GCS_MAVLink/GCS.h>
 
+#if USE_SATTELITE
+#include "satellite_telemetry.h"
+#else
+#include "low_bandwidth_telemetry.h"
+#endif
+
 #include <AP_Declination/AP_Declination.h> // ArduPilot Mega Declination Helper Library
 
 class Rover {
 public:
     friend class GCS_MAVLINK;
     friend class Parameters;
+    friend class IridiumTelemtry;
 
     Rover(void);
 
@@ -178,6 +185,12 @@ private:
     const uint8_t num_gcs;
     GCS_MAVLINK gcs[MAVLINK_COMM_NUM_BUFFERS];
 
+#if USE_SATTELITE
+    IridiumTelemtry	m_satellite_telemetry;
+#else
+    LowBandwidthTelemetry m_slow_telemetry;
+#endif
+
     // a pin for reading the receiver RSSI voltage. The scaling by 0.25 
     // is to take the 0 to 1024 range down to an 8 bit range for MAVLink
     AP_HAL::AnalogSource *rssi_analog_source;
@@ -510,6 +523,8 @@ private:
     bool in_stationary_loiter(void);
     void set_loiter_active(const AP_Mission::Mission_Command& cmd);
 
+    void satellite_telemetry(void);
+
 public:
     bool print_log_menu(void);
     int8_t dump_log(uint8_t argc, const Menu::arg *argv);
