diff --git a/APMrover2/Rover.h b/APMrover2/Rover.h
index e2224ea..718d750 100644
--- a/APMrover2/Rover.h
+++ b/APMrover2/Rover.h
@@ -93,12 +93,15 @@
 #include "Parameters.h"
 #include <GCS_MAVLink/GCS.h>
 
+#include "satellite_telemetry.h"
+
 #include <AP_Declination/AP_Declination.h> // ArduPilot Mega Declination Helper Library
 
 class Rover {
 public:
     friend class GCS_MAVLINK;
     friend class Parameters;
+    friend class IridiumTelemtry;
 
     Rover(void);
 
@@ -173,6 +176,9 @@ private:
     SITL sitl;
 #endif
 
+    IridiumTelemtry	m_satellite_telemetry;
+    GCS_MAVLINK m_satellite_gcs;
+
     // GCS handling
     AP_SerialManager serial_manager;
     const uint8_t num_gcs;
@@ -510,6 +516,8 @@ private:
     bool in_stationary_loiter(void);
     void set_loiter_active(const AP_Mission::Mission_Command& cmd);
 
+    void satellite_telemetry(void);
+
 public:
     bool print_log_menu(void);
     int8_t dump_log(uint8_t argc, const Menu::arg *argv);
